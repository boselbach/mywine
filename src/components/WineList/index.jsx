import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import WineItem from '../WineItem';
import {} from './style.scss';

class WineList extends PureComponent {
  static propTypes = {
    wineItems: PropTypes.array.isRequired
  }

  static defaultProps = {
    wineItems: []
  }

  render() {
    const { 
      wineItems,
      edit,
      remove
    } = this.props;

    return (
      <div className="wine-items">
        {wineItems.map((item, index) => (
          <WineItem
            key={index}
            item={item}
            edit={edit}
            remove={this.props.remove}
          />
        ))}
      </div>
    )
  }
}

export default WineList;