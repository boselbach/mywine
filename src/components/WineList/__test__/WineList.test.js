import React from 'react';
import { shallow } from 'enzyme';
import WineList from '../';

describe('WineList', () => {
  it('renders correctly', () => {
    const component = shallow(
      <WineList />
    );

    expect(component).toMatchSnapshot();
  });
});
