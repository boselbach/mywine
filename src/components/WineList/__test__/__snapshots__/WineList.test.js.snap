// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WineList renders correctly 1`] = `<WineItem />`;
